ext/ddtrace_profiling_native_extension/heap_recorder.c

```diff
@@ -101,6 +103,13 @@ typedef struct {
   live_object_data object_data;
 } partial_heap_recording;
 
+typedef struct {
+  // Has ownership of this, needs to clean-it-up if not transferred.
+  heap_record *heap_record;
+  long obj_id;
+  live_object_data object_data;
+} uncommitted_sample;
+
 struct heap_recorder {
   // Map[key: heap_record_key*, record: heap_record*]
   // NOTE: We always use heap_record_key.type == HEAP_STACK for storage but support lookups
```

Comment:
Was already like this before, but while going through this PR, I realized that actually the comment on `records_mutex` is innacurate -- it actually is only really projecting the `object_records`, since for the `heap_records` we rely on the GVL instead right now.

Reply:
Nice flag, fixed it upstream.